### ISSUE TITLE
IAtomEntry.Description is null for Atom entries with xhtml and */xml content types

### ISSUE THREAD

**Issue by dpypin** [dotnet/SyndicationFeedReaderWriter#6]
I tried to read a sample feed with one entry from https://tools.ietf.org/html/rfc4287#section-1.1 (second sample) and Description field of the read Atom entry was null. As far as I understand this field must contant text representation of the xhtml content without wrapper div. I tried to do the same for application/xml content type and the result was the same - Description field was null.

Repro:
```
static void AtomTest()
{
    var xhtml = "<feed xmlns=\"http://www.w3.org/2005/Atom\"><title type=\"text\">dive into mark</title><subtitle type=\"html\">A &lt;em&gt;lot&lt;/em&gt; of effort went into making this effortless</subtitle><updated>2005-07-31T12:29:29Z</updated><id>tag:example.org,2003:3</id><link rel=\"alternate\" type=\"text/html\" hreflang=\"en\" href=\"http://example.org/\"/><link rel=\"self\" type=\"application/atom+xml\" href=\"http://example.org/feed.atom\"/><rights>Copyright (c) 2003, Mark Pilgrim</rights><generator uri=\"http://www.example.com/\" version=\"1.0\">Example Toolkit</generator><entry><title>Atom draft-07 snapshot</title><link rel=\"alternate\" type=\"text/html\" href=\"http://example.org/2005/04/02/atom\"/><link rel=\"enclosure\" type=\"audio/mpeg\" length=\"1337\" href=\"http://example.org/audio/ph34r_my_podcast.mp3\"/><id>tag:example.org,2003:3.2397</id><updated>2005-07-31T12:29:29Z</updated><published>2003-12-13T08:29:29-04:00</published><author><name>Mark Pilgrim</name><uri>http://example.org/</uri><email>f8dy@example.com</email></author><contributor><name>Sam Ruby</name></contributor><contributor><name>Joe Gregorio</name></contributor><content type=\"xhtml\" xml:lang=\"en\" xml:base=\"http://diveintomark.org/\"><div xmlns=\"http://www.w3.org/1999/xhtml\"><p><i>[Update: The Atom draft is finished.]</i></p></div></content></entry></feed>";

    var feed = new List<object>();

    var byteArray = Encoding.UTF8.GetBytes(xhtml);

    using (var stream = new MemoryStream(byteArray))
    using (var xmlReader = XmlReader.Create(stream, new XmlReaderSettings { CloseInput = true }))
    {
        var feedReader = new AtomFeedReader(xmlReader);

        while (feedReader.Read().Result)
        {
            switch (feedReader.ElementType)
            {
                case SyndicationElementType.Category:
                    ISyndicationCategory category = feedReader.ReadCategory().Result;
                    feed.Add(category);
                    break;

                //
                // Read image
                case SyndicationElementType.Image:
                    ISyndicationImage image = feedReader.ReadImage().Result;
                    feed.Add(image);
                    break;

                //
                // Read entry 
                case SyndicationElementType.Item:
                    IAtomEntry entry = feedReader.ReadEntry().Result;
                    feed.Add(entry);
                    break;

                //
                // Read link
                case SyndicationElementType.Link:
                    ISyndicationLink link = feedReader.ReadLink().Result;
                    feed.Add(link);
                    break;

                //
                // Read person
                case SyndicationElementType.Person:
                    ISyndicationPerson person = feedReader.ReadPerson().Result;
                    feed.Add(person);
                    break;

                //
                // Read content
                default:
                    ISyndicationContent content = feedReader.ReadContent().Result;
                    feed.Add(content);
                    break;
            }
        }
    }

    var description = ((ISyndicationItem) feed[8]).Description; // feed[8] is the entry and description is null
}
```

**Comment by drago-draganov** [comment]
@dpypin,
Thanks for reporting this.

It's bug in XHTML content type handling in regards of Atom spec. It wasn't discovered due to flawed test feed. We will address this shortly. Here is a workaround using CDATA instead:

```
<content type="xhtml" xml:lang="en" xml:base="http://diveintomark.org/">
      <![CDATA[
      <div>
        <p>
          <i>[Update: The Atom draft is finished.]</i>
        </p>
      </div>
      ]]>
</content>
```

**Comment by dpypin** [comment]
Thanks @drago-draganov, do you have any estimations on how much time it will take to fix this bug?

**Comment by drago-draganov** [comment]
#7 pull request has the fix.

**Comment by dpypin** [comment]
Thank you for a quick response.
I checked #7 and it seems that XHTML support is there (I tested only reading, not writing), but it doesn't add support for xml and other xml-like content types.

**Comment by drago-draganov** [comment]
Atom Text Construct is limited to text, html and xhtml.
atom:content element supports the above plus any MIME type (as text) or with linked resource. In this case the content is empty and the src attribute points to resource with MIME type specified in type attribute.
The AtomParser won't set Entry Description if the content has src attribute, but instead will add a ISyndicationLink with proper MediaType.

Is there a particular example you have in mind?

**Comment by dpypin** [comment]
This paragraph says that content type can be an XML Media Type - https://tools.ietf.org/html/rfc4287#section-4.1.3.3.
This is a sample with application/xml content type where IAtomEntry.Description is null:
```
<feed xmlns="http://www.w3.org/2005/Atom">
    <title>Example Feed</title>
    <link href="http://example.org/"/>
    <updated>2003-12-13T18:30:02Z</updated>
    <author>
        <name>John Doe</name>
    </author>
    <id>urn:uuid:60a76c80-d399-11d9-b93C-0003939e0af6</id>
    <entry>
        <title>Atom-Powered Robots Run Amok</title>
        <link href="http://example.org/2003/12/13/atom03"/>
        <id>urn:uuid:1225c695-cfb8-4ebb-aaaa-80da344efa6a</id>
        <updated>2003-12-13T18:30:02Z</updated>
        <summary>Some text.</summary>
        <content type="application/xml">
            <tag1 attr1="123">
                234
            </tag1>
        </content>
    </entry>
</feed>
```

**Comment by drago-draganov** [comment]
Agreed. Tracking via #9 
Thanks!

**Comment by drago-draganov** [comment]
Fixed in #7 and #12
